### PR TITLE
feat: Argo Workflow blueprint upgrade to latest Karpenter

### PR DESCRIPTION
**Pull request by ovaleanu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Upgrade Karpenter version for Argo Workflow blueprint

### Motivation

[#413 ](https://github.com/awslabs/data-on-eks/issues/413)

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: schedulers/terraform/argo-workflow/karpenter-provisioners/spark-compute-optimized-provisioner.yaml [hunk 1]
```diff
@@ -1,109 +1,111 @@
-apiVersion: karpenter.sh/v1alpha5
```
Comment (vara-bonthu) [inline schedulers/terraform/argo-workflow/karpenter-provisioners/spark-compute-optimized-provisioner.yaml:1]: This is new way of deploying Karpenter provisioners which uses a local Helm Chart from EKS Data Add-ons
https://github.com/awslabs/data-on-eks/blob/1e4b51971bed042adf17cb7995cceafb7790a150/analytics/terraform/spark-k8s-operator/addons.tf#L180C1-L184C1

If you have bandwidth, then you can leverage the above approach or raise another PR for the upgrade. I am fine with either approach.

File: schedulers/terraform/argo-workflow/addons.tf [hunk 2]
```diff
@@ -113,7 +113,15 @@ module "eks_blueprints_addons" {
   #---------------------------------------
   enable_karpenter                  = true
   karpenter_enable_spot_termination = true
+  karpenter_node = {
+    iam_role_use_name_prefix = false
+    iam_role_name            = "${local.name}-karpenter-node"
+    iam_role_additional_policies = {
+      AmazonSSMManagedInstanceCore = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+    }
+  }
```
Comment (vara-bonthu) [inline schedulers/terraform/argo-workflow/addons.tf:122]: Check this code where we are not defining `iam_role_name`. 

https://github.com/awslabs/data-on-eks/blob/1e4b51971bed042adf17cb7995cceafb7790a150/analytics/terraform/spark-k8s-operator/addons.tf#L82C1-L96C4

iam role is being passed to the Karpenter provisioner here https://github.com/awslabs/data-on-eks/blob/1e4b51971bed042adf17cb7995cceafb7790a150/analytics/terraform/spark-k8s-operator/addons.tf#L189

**Review by vara-bonthu — COMMENTED** [review]
@ovaleanu Thanks for the upgrade 👍🏼  I left couple of comments .

File: schedulers/terraform/argo-workflow/addons.tf (hunk 2 above)
Comment (ovaleanu) [reply schedulers/terraform/argo-workflow/addons.tf:122]: I saw this, but if the provisioner is in a file separately and not embedded in the addons.tf, then it doesn't know who is "module".

File: schedulers/terraform/argo-workflow/karpenter-provisioners/spark-compute-optimized-provisioner.yaml (hunk 1 above)
Comment (ovaleanu) [reply schedulers/terraform/argo-workflow/karpenter-provisioners/spark-compute-optimized-provisioner.yaml:1]: I have seen this as well. I can use this approach.

File: schedulers/terraform/argo-workflow/addons.tf [hunk 3]
```diff
@@ -234,25 +240,133 @@ module "eks_data_addons" {
...
+# eks_cluster_id = module.eks.cluster_name
+# }
+# }
+#
+# resource "kubectl_manifest" "karpenter_provisioner" {
+# for_each  = toset(data.kubectl_path_documents.karpenter_provisioners.documents)
+# yaml_body = each.value
+#
+# depends_on = [module.eks_blueprints_addons]
+# }
```
Comment (vara-bonthu) [inline schedulers/terraform/argo-workflow/addons.tf:370]: you can remove the commented code block

Comment (vara-bonthu) [reply schedulers/terraform/argo-workflow/addons.tf:370]: kubectl provider may not be required as well if you are not using any kubectl_manifest commands

**Review by vara-bonthu — COMMENTED** [review]
Thanks for updating the PR. Please remove the commented code so that we can merge the PR

File: schedulers/terraform/argo-workflow/addons.tf (hunk 3 above)
Comment (ovaleanu) [reply schedulers/terraform/argo-workflow/addons.tf:370]: Fixed. Thank you!

**Review by vara-bonthu — APPROVED** [review]
🚀

**Pull request merged**